ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/JaasAuthProvider.java

```diff
@@ -128,7 +130,16 @@ private void getUser(
     // We do the actual authorization here not in the User class
     final boolean authorized = validateRoles(lc, allowedRoles);
 
-    promise.complete(new JaasUser(username, password, authorized));
+    // if the subject from the login context is already a KsqlPrincipal, use the subject
+    // directly rather than creating a new one
+    final Optional<KsqlPrincipal> ksqlPrincipal = lc.getSubject().getPrincipals().stream()
+        .filter(p -> p instanceof KsqlPrincipal)
+        .map(p -> (KsqlPrincipal)p)
+        .findFirst();
+    final JaasUser user = ksqlPrincipal.isPresent()
+        ? new JaasUser(ksqlPrincipal.get(), authorized)
+        : new JaasUser(username, password, authorized);
```

Comment:
The password is used in the confluent-security-plugins to create the impersonated user context. It is important to make autho. requests with the basic credentials.

If you plan to add it as a required param in `KsqlPrincipal`, then does it make sense to delete the `JaasPrincipal`? It won't be used anymore as KsqlPrincipal will have the same info.

Reply:
We still need to keep some KsqlPrincipal implementation in the ksql repo, since users can use basic authN without a custom auth plugin that returns a KsqlPrincipal. For this reason, I think we still want to keep JaasPrincipal.